### PR TITLE
Simplify update_basket task.

### PR DESCRIPTION
**Pull request by akatsoulas**

### CODE REVIEW

**Comment by johngian** [comment]
Please fix typo in commit message/PR title

File: mozillians/users/tasks.py [hunk 1]
```diff
@@ -102,18 +103,13 @@ def update_basket_task(instance_id, newsletters=[]):
...
-            except (MaxRetriesExceededError, basket.BasketException):
-                msg = exception.message
-                _email_basket_managers('update_phonebook', token, msg)
-            return
+            retry = True
 
         # Check if the users needs to be subscribed to additional newsletters.
         newsletters_to_subscribe = [nl for nl in newsletters if nl not in result['newsletters']]
 
         old_email = result['email']
```
Comment (johngian) [inline mozillians/users/tasks.py]: When ``retry==True`` ``result`` var is undefined.

**Comment by johngian** [comment]
Instead of having these endless ``if/else/elif`` cases i suggest we wrap all the basket calls to a helper method that takes as arguments:

* basket method
* basket method args
* task

and does the heavy-lifting of celery retries.

**Comment by akatsoulas** [comment]
Closing this as it's not built in the right direction. I will open a new one for this.

**Pull request closed without merging**